New layout to support evergreen campaigns / weekly budget

j6ll opened the pull request:
Related to #



## Proposed Changes
**Before**
![Screenshot 2024-03-11 at 15 57 48](https://github.com/Automattic/wp-calypso/assets/6440498/8d9778c8-1bb2-4e66-a644-c3213cdea50b)

**After (left traditional, right evergreen)**
![Screenshot 2024-03-11 at 15 57 08](https://github.com/Automattic/wp-calypso/assets/6440498/f9d14d0a-c008-4259-9245-53fb91974718)

**Design**
![Screenshot 2024-03-12 at 16 10 18](https://github.com/Automattic/wp-calypso/assets/6440498/b29e7cb1-280c-4093-9768-bc64cc5f949a)


## Testing Instructions

Check the page using the live  link below, this should let you check a traditional campaign.

To test an evergreen campaign, you will need to load calypso locally, then force `is_evergreen` to `true`, see this commit for an example:
https://github.com/Automattic/wp-calypso/pull/88379/commits/ee1428a577ed6590efc33324c772feffb29b5d9c


Note: 

- "Duration so far" will only show when campaign is active too.
- Weekly spend summary will be in a follow on PR, I need to add this info to the campaign endpoint
- I've hidden stats when evergreen campaign is scheduled, Since "spent so far" values are all empty and so is impressions
![Screenshot 2024-03-12 at 17 33 11](https://github.com/Automattic/wp-calypso/assets/6440498/375ecd71-d1ac-46fc-b53a-64848945f9bf)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

j6ll committed:
New layout to support evergreen campaigns / weekly budget

j6ll committed:
remove dupe const

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11592735

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @j6ll for including a screenshot in the description! This is really helpful for our translators.

j6ll committed:
this should be false by default

j6ll committed:
Hide the stats when a scheduled evergreen, they don't make sense

j6ll committed:
Adds orders to the campaigns overview page under "weekly spend"

j6ll committed:
make the order rows more compact, like design

j6ll committed:
Adds some mobile styles

j6ll committed:
Revert hiding zero stats

therocket-gr commented:
I would suggest to add the 
`{ ! isLoading ? ( content here ) : <FlexibleSkeleton /> }`
to the newly added sections.. so that we have a proper loading state as we do in other sections

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11592735).

timur987 commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -595,15 +576,76 @@ export default function CampaignItemDetails( props: Props ) {
-						) }
+							) }
+
+							{ ! is_evergreen && shouldShowStats && (
+								<div className="campaign-item-details__main-stats-row">
+									<div>
+										<span className="campaign-item-details__label">
+											{ is_evergreen && status === 'active'
```
@j6ll, I'm in doubt how `is_evergreen` can be `true` (line `585`) inside the condition (line `581`) of `! is_evergreen`.

j6ll replied:
Thanks timur, good spot, I made a tweak and forgot to remove this statement, I will update!

j6ll replied:
Ah, sorry this is a bit messed up.  I actually overwrite all this logic in the next PR #88487.

So if we ignore for now, I plan to merge both branches at the same

j6ll replied:
Thanks for spotting though, completely right :)

timur987 commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -595,15 +576,76 @@ export default function CampaignItemDetails( props: Props ) {
+											{ ! isLoading ? durationFormatted : <FlexibleSkeleton /> }
+										</span>
+									</div>
+									{ is_evergreen ? (
```
And here as well.

j6ll replied:
Thanks!

timur987 reviewed (commented):
@j6ll, I tried to test this PR using local Calypso.

Testing instructions were a bit confusing for me but I was able to repeat the last image you've provided.

(Your version)

![image](https://private-user-images.githubusercontent.com/6440498/312189265-375ecd71-d1ac-46fc-b53a-64848945f9bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3NDc5OTEsIm5iZiI6MTcxMDc0NzY5MSwicGF0aCI6Ii82NDQwNDk4LzMxMjE4OTI2NS0zNzVlY2Q3MS1kMWFjLTQ2ZmMtYjUzYS02NDg0ODk0NWY5YmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMThUMDc0MTMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWQwMzI3NWQwNGQ4MzViMTY2NTNiMzQ3ZTI4MjdiMGFiOTEzNWI5MDliNDliNGZmMGZjMGNhYzM0MDMyZTk0MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2nQkeM5zoN5i2XCtVxn8lkLLs61F1kLmegj7bhJ2rYw)

(My version)

<img width="981" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/836e86c3-25a7-49fc-b646-60b6769515c4">

I left a couple of comments though.

I tried to imitate what you've depicted in the first three images but (as I've mentioned before) only repeated the last image.

If there's a way to create fake impressions in a local DB to be able to repeat all images from the description, would be happy to know how to do it and test again. Of course, if the only intention was to reproduce the last image, will be happy to approve this PR.

j6ll commented:
> Testing instructions were a bit confusing for me but I was able to repeat the last image you've provided.
Sorry amigo and yes they are a bit vague, I didn't want to give specifics in a public repo. But I'm glad you got sorted.

> If there's a way to create fake impressions in a local DB to be able to repeat all images from the description, would be happy to know how to do it and test again.

Sure, this comes from `dsp_campaigns` table so you can add the data there.

therocket-gr reviewed (approved):
tested locally.. LGTM

j6ll committed:
Revert hiding for evergreen

j6ll committed:
Fixes:
Calculate campaign total, since orders have multiple campaigns
Fix duration days and order dates
Don't allow budget below zero
Fixes remaining being / 100W
Ordering for non evergreen campaigns
Promo codes not included

timur987 reviewed (approved):
Thanks, @j6ll!

The code LGTM and works as expected.

j6ll commented:
Janky rebase, means this branch now includes https://github.com/Automattic/wp-calypso/pull/88487, no problem though I wanted to ship both at the same time.
